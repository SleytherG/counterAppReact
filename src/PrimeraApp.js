import React from "react";
import PropTypes from 'prop-types';
/**
 * Functional Components
 */
const PrimeraApp = ( { saludo, subtitulo } ) => {

    return (
        <React.Fragment>
            <h1> { saludo }!!!</h1>
            {/*<pre> { JSON.stringify( saludo, null, 3 ) } </pre>*/}
            <p id="texto">{ subtitulo }</p>
        </React.Fragment>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un Subtitulo'
}

export default PrimeraApp;
