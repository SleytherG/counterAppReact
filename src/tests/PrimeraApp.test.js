import React from "react";
import {render} from "@testing-library/react";
import '@testing-library/jest-dom'
import PrimeraApp from "../PrimeraApp";
import {shallow} from "enzyme";

describe('Pruebas en <PrimeraApp />', () => {

    // test('Debe de mostrar el mensaje Hola soy goku', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render( <PrimeraApp saludo={ 'Hola, soy Goku' } />);
    //     expect( getByText( saludo + '!!' ) ).toBeInTheDocument();
    // });

    test('Debe de mostrar el componente <PrimeraApp /> Correctamente', () => {

        const saludo = 'Hola ,soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/>);

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Soy un Subtitulo';

        const wrapper = shallow(<PrimeraApp saludo={ saludo } subtitulo={ subtitulo }/>);

        const textoParrafo = wrapper.find('p').text();

        console.log(textoParrafo);

        expect( textoParrafo ).toBe( subtitulo );
    })
})
