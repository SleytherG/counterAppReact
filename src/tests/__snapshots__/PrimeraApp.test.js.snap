// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar el componente <PrimeraApp /> Correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola ,soy Goku
    !!!
  </h1>
  <p
    id="texto"
  >
    Soy un Subtitulo
  </p>
</Fragment>
`;
